bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnectionConfig.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.transport.mqtt;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Contains configuration for a MqttBrokerConnection. Necessary to add a new broker connection the {@link MqttService}.
+ *
+ * @author David Graeff - Initial contribution and API
+ */
+public class MqttBrokerConnectionConfig {
+    // Optional connection name
+    public @Nullable String name;
+    // Connection parameters (host+port+secure)
+    public @Nullable String host;
+    public @Nullable Integer port;
+    public Boolean secure = true;
+    // Authentication parameters
+    public @Nullable String username;
+    public @Nullable String password;
+    public @Nullable String clientID;
+    // MQTT parameters
+    public Integer qos = MqttBrokerConnection.DEFAULT_QOS;
+    public Boolean retainMessages = false;
+    /** Keepalive in seconds */
+    public @Nullable Integer keepAlive;
+    // Last will parameters
+    public @Nullable String lwtTopic;
```

Comment:
Now the "old" configuration of lwt is not supported anymore. With https://github.com/eclipse/smarthome/pull/5231 @maggu2810 made sure both options could be used.
I thinks its fine that the UI (config description) only provides the single fields. But text based configuration may depend on the single line config. wdyt?

Reply:
Usually yes, it would be a good idea to preserve old fields. But: This PR is a API brake, code-wise as well as configuration-wise. Remember that we used the dot syntax before to express multiple mqtt instances and we use osgi service factories with a config file each now. Any user of this service (which I doubt there are any, because we provide no means to use Mqtt really) need to migrate anyway ^^.

Reply:
@maggu2810 fine for you?

Reply:
It is okay for me to drop the old way to configure the LWT.

I am pretty sure there are guys the use this service. There has been one that inspired me to implement that change (to support a more generic topic and payload) and I am using that service, too. Already two of them. :wink: But as @davidgraeff also pointed out, changes in its usage needs to be done regardless if we keep this field or not, so I would prefer to remove old stuff and stop maintaining it on this change and all the other ones that will come some time.